plc4j/tools/connection-cache/src/main/java/org/apache/plc4x/java/utils/cache/LeasedPlcConnection.java

```diff
@@ -51,13 +51,23 @@ public void run() {
 
     @Override
     public synchronized void close() {
-        // Make the connection unusable.
-        connection = null;
-
+        if(connectionContainer == null) {
+            return;
+        }
```

Comment:
Why should the connectionContainer ever be null? The way I understand it, the only way this could happen, would be that a program got a lease ... executed a request, that had an issue and therefore it automatically closed the connection and destroy() called close from the inside and then the using application calls close regularly ... not too happy with this workflow in general ... we should really discuss things on the list.

Reply:
Close twice. eg: user close this connection but usageTimer will close it again. It will cause nullpointexception

Reply:
Yeah ... when returning a connection after a connection timed-out or got another error, it would have thrown an error from inside the returnConnection method of the Container.

Reply:
This is a bug.  It should not throw an error. If you change usageTimer from local variable to private variable i can canel the Time. But...  I still think the Timer is not necessary. So i don't do anyting. Just add this code.